---
title: "Normalisation_subseting_and_merging_of_mouse_OB"
output: html_document
params:

  sample_name1: A2_OB_virgin
  file_dir1: #Dir to sample from space_ranger output /A2_OB_virgin/outs/
  
  sample_name2: B2_OB_mother_20
  file_dir2: #Dir to sample from space_ranger output /B2_OB_mother_20/outs/
  
  sample_name3: C2_OB_virgin_30
  file_dir3: #Dir to sample from space_ranger output /C2_OB_virgin_30/outs/
  
  sample_name4: D2_OB_mother_30
  file_dir4: #Dir to sample from space_ranger output /D2_OB_mother_30/outs/
  
  raw_UMI_counts: raw_feature_bc_matrix.h5
  filtered_UMI_counts: filtered_feature_bc_matrix.h5
  imgs: spatial/tissue_hires_image.png
  spotfiles: spatial/tissue_positions_list.csv
  json: spatial/scalefactors_json.json
  
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r 002, librarys, data loading, message=FALSE,}

library(Seurat)
#library(SeuratData)
library(ggplot2)
library(patchwork)
library(dplyr)
library(rmarkdown)
library(ggrepel)

set.seed(42)

sample_name1 <- params$sample_name1
file_dir1 <- params$file_dir1

sample_name2 <- params$sample_name2
file_dir2 <- params$file_dir2

sample_name3 <- params$sample_name3
file_dir3 <- params$file_dir3

sample_name4 <- params$sample_name4
file_dir4 <- params$file_dir4

filtered_UMI_counts <- params$filtered_UMI_counts
raw_UMI_counts <- params$raw_UMI_counts
imgs <- params$imgs
spotfiles <- params$spotfiles
json <- params$json
```

Adappted from:https://rdrr.io/github/satijalab/seurat/src/R/preprocessing.R#sym-Load10X_Spatial
to name SeuratProject
```{r 003, load10x adapted with project, message=FALSE,}
#to name project 
Load10X_Spatial_proj <- function(
  data.dir,
  filename = 'filtered_feature_bc_matrix.h5',
  assay = 'Spatial',
  slice = 'slice1',
  filter.matrix = TRUE,
  to.upper = FALSE,
  project = 'SeuratProject',
  ...
) {
  if (length(x = data.dir) > 1) {
    warning("'Load10X_Spatial' accepts only one 'data.dir'", immediate. = TRUE)
    data.dir <- data.dir[1]
  }
  data <- Read10X_h5(filename = file.path(data.dir, filename), ...)
  if (to.upper) {
    rownames(x = data) <- toupper(x = rownames(x = data))
  }
  object <- CreateSeuratObject(counts = data, project = project, assay = assay)
  image <- Read10X_Image(
    image.dir = file.path(data.dir, 'spatial'),
    filter.matrix = filter.matrix
  )
  image <- image[Cells(x = object)]
  DefaultAssay(object = image) <- assay
  object[[slice]] <- image
  return(object)
}
```


```{r 004, data loading, message=FALSE,}

spat.data_A<-Load10X_Spatial_proj(data.dir = file_dir1,filename = filtered_UMI_counts, project = "A2_OB_virgin20")

spat.data_B<-Load10X_Spatial_proj(data.dir = file_dir2,filename = filtered_UMI_counts, project = "B2_OB_mother20")

spat.data_C<-Load10X_Spatial_proj(data.dir = file_dir3,filename = filtered_UMI_counts, project = "C2_OB_virgin30")

spat.data_D<-Load10X_Spatial_proj(data.dir = file_dir4,filename = filtered_UMI_counts, project = "D2_OB_mother30")

```

# Quality control
Gene expression is high in all the samples 



```{r 005, data loading, message=FALSE,}
p1 <- ggplot() +
  geom_histogram(data = spat.data_A[[]], aes(nFeature_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Unique genes per spot")

p2 <- ggplot() +
  geom_histogram(data = spat.data_A[[]], aes(nCount_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Total counts per spots")

gene_attr <- data.frame(nUMI = Matrix::rowSums(spat.data_A@assays$Spatial@counts), 
                        nSpots = Matrix::rowSums(spat.data_A@assays$Spatial@counts > 0))
p3 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nUMI), fill = "red", alpha = 0.7, bins = 50) +
  scale_x_log10() +
  ggtitle("Total counts per gene (log10 scale)")

p4 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nSpots), fill = "red", alpha = 0.7,  bins = 50) +
  ggtitle("Total spots per gene")

(p1 - p2)/(p3 - p4)
###########################################################################################
p5 <- ggplot() +
  geom_histogram(data = spat.data_B[[]], aes(nFeature_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Unique genes per spot")

p6 <- ggplot() +
  geom_histogram(data = spat.data_B[[]], aes(nCount_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Total counts per spots")

gene_attr <- data.frame(nUMI = Matrix::rowSums(spat.data_B@assays$Spatial@counts), 
                        nSpots = Matrix::rowSums(spat.data_B@assays$Spatial@counts > 0))
p7 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nUMI), fill = "red", alpha = 0.7, bins = 50) +
  scale_x_log10() +
  ggtitle("Total counts per gene (log10 scale)")

p8 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nSpots), fill = "red", alpha = 0.7,  bins = 50) +
  ggtitle("Total spots per gene")

(p5 - p6)/(p7 - p8)


###########################################################################################
p9 <- ggplot() +
  geom_histogram(data = spat.data_C[[]], aes(nFeature_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Unique genes per spot")

p10 <- ggplot() +
  geom_histogram(data = spat.data_C[[]], aes(nCount_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Total counts per spots")

gene_attr <- data.frame(nUMI = Matrix::rowSums(spat.data_C@assays$Spatial@counts), 
                        nSpots = Matrix::rowSums(spat.data_C@assays$Spatial@counts > 0))
p11 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nUMI), fill = "red", alpha = 0.7, bins = 50) +
  scale_x_log10() +
  ggtitle("Total counts per gene (log10 scale)")

p12 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nSpots), fill = "red", alpha = 0.7,  bins = 50) +
  ggtitle("Total spots per gene")

(p9 - p10)/(p11 - p12)
###########################################################################################
p13 <- ggplot() +
  geom_histogram(data = spat.data_D[[]], aes(nFeature_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Unique genes per spot")

p14 <- ggplot() +
  geom_histogram(data = spat.data_D[[]], aes(nCount_Spatial), fill = "red", alpha = 0.7, bins = 50) +
  ggtitle("Total counts per spots")

gene_attr <- data.frame(nUMI = Matrix::rowSums(spat.data_D@assays$Spatial@counts), 
                        nSpots = Matrix::rowSums(spat.data_D@assays$Spatial@counts > 0))
p15 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nUMI), fill = "red", alpha = 0.7, bins = 50) +
  scale_x_log10() +
  ggtitle("Total counts per gene (log10 scale)")

p16 <- ggplot() +
  geom_histogram(data = gene_attr, aes(nSpots), fill = "red", alpha = 0.7,  bins = 50) +
  ggtitle("Total spots per gene")

(p13 - p14)/(p15 - p16)



```

## % of mitochondrial genes and ribosmoal protein


```{r 006, data loading, message=FALSE,}

# Collect all genes coded on the mitochondrial genome
mt.genes <- grep(pattern = "^mt-", x = rownames(spat.data_A), value = TRUE)
spat.data_A$percent.mito <- (Matrix::colSums(spat.data_A@assays$Spatial@counts[mt.genes, ])/Matrix::colSums(spat.data_A@assays$Spatial@counts))*100

# Collect all genes coding for ribosomal proteins
rp.genes <- grep(pattern = "^Rpl|^Rps",x = rownames(spat.data_A), value = TRUE)
spat.data_A$percent.ribo <- (Matrix::colSums(spat.data_A@assays$Spatial@counts[rp.genes, ])/Matrix::colSums(spat.data_A@assays$Spatial@counts))*100


SpatialFeaturePlot(spat.data_A, features = "percent.mito", pt.size.factor = 1)
SpatialFeaturePlot(spat.data_A, features = "percent.ribo", pt.size.factor = 1)
####################################################################################
# Collect all genes coded on the mitochondrial genome
mt.genes <- grep(pattern = "^mt-", x = rownames(spat.data_B), value = TRUE)
spat.data_B$percent.mito <- (Matrix::colSums(spat.data_B@assays$Spatial@counts[mt.genes, ])/Matrix::colSums(spat.data_B@assays$Spatial@counts))*100

# Collect all genes coding for ribosomal proteins
rp.genes <- grep(pattern = "^Rpl|^Rps",x = rownames(spat.data_B), value = TRUE)
spat.data_B$percent.ribo <- (Matrix::colSums(spat.data_B@assays$Spatial@counts[rp.genes, ])/Matrix::colSums(spat.data_B@assays$Spatial@counts))*100


SpatialFeaturePlot(spat.data_B, features = "percent.mito", pt.size.factor = 1)
SpatialFeaturePlot(spat.data_B, features = "percent.ribo", pt.size.factor = 1)
#######################################################################################
# Collect all genes coded on the mitochondrial genome
mt.genes <- grep(pattern = "^mt-", x = rownames(spat.data_C), value = TRUE)
spat.data_C$percent.mito <- (Matrix::colSums(spat.data_C@assays$Spatial@counts[mt.genes, ])/Matrix::colSums(spat.data_C@assays$Spatial@counts))*100

# Collect all genes coding for ribosomal proteins
rp.genes <- grep(pattern = "^Rpl|^Rps",x = rownames(spat.data_C), value = TRUE)
spat.data_C$percent.ribo <- (Matrix::colSums(spat.data_C@assays$Spatial@counts[rp.genes, ])/Matrix::colSums(spat.data_C@assays$Spatial@counts))*100


SpatialFeaturePlot(spat.data_C, features = "percent.mito", pt.size.factor = 1)
SpatialFeaturePlot(spat.data_C, features = "percent.ribo", pt.size.factor = 1)

#######################################################################################
# Collect all genes coded on the mitochondrial genome
mt.genes <- grep(pattern = "^mt-", x = rownames(spat.data_D), value = TRUE)
spat.data_D$percent.mito <- (Matrix::colSums(spat.data_D@assays$Spatial@counts[mt.genes, ])/Matrix::colSums(spat.data_D@assays$Spatial@counts))*100

# Collect all genes coding for ribosomal proteins
rp.genes <- grep(pattern = "^Rpl|^Rps",x = rownames(spat.data_D), value = TRUE)
spat.data_D$percent.ribo <- (Matrix::colSums(spat.data_D@assays$Spatial@counts[rp.genes, ])/Matrix::colSums(spat.data_D@assays$Spatial@counts))*100


SpatialFeaturePlot(spat.data_D, features = "percent.mito", pt.size.factor = 1)
SpatialFeaturePlot(spat.data_D, features = "percent.ribo", pt.size.factor = 1)


```

# Visualisation of Gene expression and Reads per spot
With this we also can see the region that were folded over each other high gene expression difficult to filter since might remove other regions. 

```{r 007, message=FALSE,}
# spat.data_A_filter <- subset(spat.data_A, nFeature_Spatial > 600 & percent.mito < 20)
# cat("Spots removed: ", ncol(spat.data_A) - ncol(spat.data_A_filter), "\n")
# #################################################################################
# spat.data_B <- subset(spat.data_B, nFeature_Spatial > 600 & percent.mito < 20)
# #cat("Spots removed: ", ncol(spat.data_B) - ncol(filter.spat.data_B), "\n")
# ################################################################################
# spat.data_C <- subset(spat.data_C, nFeature_Spatial > 600 & percent.mito < 20)
# #cat("Spots removed: ", ncol(spat.data_C) - ncol(filter.spat.data_C), "\n")
# ################################################################################
# spat.data_D <- subset(spat.data_D, nFeature_Spatial > 600 & percent.mito < 20)
# #cat("Spots removed: ", ncol(spat.data_D) - ncol(filter.spat.data_D), "\n")

SpatialFeaturePlot(spat.data_A, features = "nFeature_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_A, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_B, features = "nFeature_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_B, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_C, features = "nFeature_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_C, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_D, features = "nFeature_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_D, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_A, features = "nCount_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_A, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_B, features = "nCount_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_B, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_C, features = "nCount_Spatial") + theme(legend.position = "right")
#SpatialFeaturePlot(filter.spat.data_C, features = "nFeature_Spatial") + theme(legend.position = "right")

SpatialFeaturePlot(spat.data_D, features = "nCount_Spatial") + theme(legend.position = "right")
```

# SCT to normalize Data 
res 0.5 for A C D and for B 0.8
this is to facilitate the subseting and remove the connective tissue and unwanted anatomical regions 

```{r 008, normalization, warning=FALSE, message=FALSE, message=FALSE,,width = 8,fig.height = 10}

spat.data_A <- SCTransform(spat.data_A, assay = "Spatial", verbose = FALSE)
spat.data_B <- SCTransform(spat.data_B, assay = "Spatial", verbose = FALSE)
spat.data_C <- SCTransform(spat.data_C, assay = "Spatial", verbose = FALSE)
spat.data_D <- SCTransform(spat.data_D, assay = "Spatial", verbose = FALSE)
spat.data_A <- RunPCA(spat.data_A, assay = "SCT", verbose = FALSE)
spat.data_B <- RunPCA(spat.data_B, assay = "SCT", verbose = FALSE)
spat.data_C <- RunPCA(spat.data_C, assay = "SCT", verbose = FALSE)
spat.data_D <- RunPCA(spat.data_D, assay = "SCT", verbose = FALSE)

#ElbowPlot(spat.data)
spat.data_A <- FindNeighbors(spat.data_A, reduction = "pca", dims = 1:20)
spat.data_A <- FindClusters(spat.data_A, verbose = FALSE,resolution = 0.5)
spat.data_A <- RunUMAP(spat.data_A, reduction = "pca", dims = 1:20)

#ElbowPlot(spat.data)
spat.data_B <- FindNeighbors(spat.data_B, reduction = "pca", dims = 1:20)
spat.data_B <- FindClusters(spat.data_B, verbose = FALSE,resolution = 0.8)
spat.data_B <- RunUMAP(spat.data_B, reduction = "pca", dims = 1:20)

#ElbowPlot(spat.data)
spat.data_C <- FindNeighbors(spat.data_C, reduction = "pca", dims = 1:20)
spat.data_C <- FindClusters(spat.data_C, verbose = FALSE,resolution = 0.5)
spat.data_C <- RunUMAP(spat.data_C, reduction = "pca", dims = 1:20)

#ElbowPlot(spat.data)
spat.data_D <- FindNeighbors(spat.data_D, reduction = "pca", dims = 1:20)
spat.data_D <- FindClusters(spat.data_D, verbose = FALSE,resolution = 0.5)
spat.data_D <- RunUMAP(spat.data_D, reduction = "pca", dims = 1:20)

# save before subset

# save(spat.data_A, file = "0.5_spat.data_A.RData")
# save(spat.data_B, file = "0.8_spat.data_B.RData")
# save(spat.data_C, file = "0.5_spat.data_C.RData")
# save(spat.data_D, file = "0.5_spat.data_D.RData")

SpatialDimPlot(spat.data_A)
SpatialDimPlot(spat.data_A, cells.highlight = CellsByIdentities(object = spat.data_A, idents = c(0, 1, 2, 3, 4, 5,6,7,8,9,10,11,12)), facet.highlight = TRUE)

SpatialDimPlot(spat.data_B)
SpatialDimPlot(spat.data_B, cells.highlight = CellsByIdentities(object = spat.data_B, idents = c(0, 1, 2, 3, 4, 5,6,7,8,9,10)), facet.highlight = TRUE)

SpatialDimPlot(spat.data_C)
SpatialDimPlot(spat.data_C, cells.highlight = CellsByIdentities(object = spat.data_C, idents = c(0, 1, 2, 3, 4, 5,6,7,8,9,10)), facet.highlight = TRUE)

SpatialDimPlot(spat.data_D)
SpatialDimPlot(spat.data_D, cells.highlight = CellsByIdentities(object = spat.data_D, idents = c(0, 1, 2, 3, 4, 5,6,7,8,9,10)), facet.highlight = TRUE)

##################################
markers_A2 <- FindAllMarkers(spat.data_A, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
markers_A2 %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(spat.data_A, features = top10$gene) + NoLegend()

markers_B2 <- FindAllMarkers(spat.data_B, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
markers_B2 %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(spat.data_B, features = top10$gene) + NoLegend()

markers_C2 <- FindAllMarkers(spat.data_C, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
markers_C2 %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(spat.data_C, features = top10$gene) + NoLegend()

markers_D2 <- FindAllMarkers(spat.data_D, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
markers_D2 %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(spat.data_D, features = top10$gene) + NoLegend()

```

# Subset

Subset is used to exclude the ONL in the mothers and virgins and to correct for anatomical differences in the surrounding tissue.


For sample A it was chosen to remove clusters 5 6 7 10
Sample B cluster 0
Sample C cluster 3 and 5 
Sample D cluster 4

The subset samples were normalized again on the "Spatial" assay (RNA assay in sc data)
processed with 20dims and res. 0.5



```{r 009, subset, warning=FALSE, message=FALSE, message=FALSE,,width = 8,fig.height = 10}
subset_A <- subset(spat.data_A, idents = c(0, 1, 2, 3, 4,8,9,11,12))
subset_B <- subset(spat.data_B, idents = c( 1, 2, 3, 4, 5,6,7,8,9,10))
subset_C <- subset(spat.data_C, idents = c( 0,1, 2, 4,6,7,8,9,10))
subset_D <- subset(spat.data_D, idents = c(0, 1, 2, 3,5,6,7,8,9,10)) 
subset_A <- SCTransform(subset_A, assay = "Spatial", verbose = FALSE)
subset_B <- SCTransform(subset_B, assay = "Spatial", verbose = FALSE)
subset_C <- SCTransform(subset_C, assay = "Spatial", verbose = FALSE)
subset_D <- SCTransform(subset_D, assay = "Spatial", verbose = FALSE)

subset_A <- RunPCA(subset_A, assay = "SCT", verbose = FALSE)
subset_B <- RunPCA(subset_B, assay = "SCT", verbose = FALSE)
subset_C <- RunPCA(subset_C, assay = "SCT", verbose = FALSE)
subset_D <- RunPCA(subset_D, assay = "SCT", verbose = FALSE)

subset_A <- FindNeighbors(subset_A, reduction = "pca", dims = 1:20)
subset_A <- FindClusters(subset_A, verbose = FALSE,resolution = 0.5)
subset_A <- RunUMAP(subset_A, reduction = "pca", dims = 1:20)
SpatialDimPlot(subset_A)

subset_B <- FindNeighbors(subset_B, reduction = "pca", dims = 1:20)
subset_B <- FindClusters(subset_B, verbose = FALSE,resolution = 0.5)
subset_B <- RunUMAP(subset_B, reduction = "pca", dims = 1:20)
SpatialDimPlot(subset_B)

subset_C <- FindNeighbors(subset_C, reduction = "pca", dims = 1:20)
subset_C <- FindClusters(subset_C, verbose = FALSE,resolution = 0.5)
subset_C <- RunUMAP(subset_C, reduction = "pca", dims = 1:20)
SpatialDimPlot(subset_C)

subset_D <- FindNeighbors(subset_D, reduction = "pca", dims = 1:20)
subset_D <- FindClusters(subset_D, verbose = FALSE,resolution = 0.5)
subset_D <- RunUMAP(subset_D, reduction = "pca", dims = 1:20)
SpatialDimPlot(subset_D)

# save subset 

# save(subset_A, file = "0.5_subset_A.RData")
# save(subset_B, file = "0.5_subset_B.RData")
# save(subset_C, file = "0.5_subset_C.RData")
# save(subset_D, file = "0.5_subset_D.RData")
```

# Merge
the enable joint dimensional reduction and clustering on the RNA expression the different slices were merged and normalized with SCT on the spatial assay
used 20 dims and for finding clusters with 0.5




```{r 010, normalization, warning=FALSE, message=FALSE, message=FALSE,}

spat.merge.all <- merge(subset_A, c(subset_B, subset_C, subset_D), add.cell.ids = c("A2", "B2","C2","D2"))


spat.merge.all <- SCTransform(spat.merge.all, assay = "Spatial", verbose = FALSE)


DefaultAssay(spat.merge.all) <- "SCT"



VariableFeatures(spat.merge.all) <- c(VariableFeatures(subset_A), VariableFeatures(subset_B), VariableFeatures(subset_C), VariableFeatures(subset_D))
spat.merge.all <- RunPCA(spat.merge.all, verbose = FALSE)
ElbowPlot(spat.merge.all)# between 10 and 15 
spat.merge.all <- FindNeighbors(spat.merge.all, dims = 1:20)
spat.merge.all <- FindClusters(spat.merge.all, verbose = FALSE,resolution = 0.5)
spat.merge.all <- RunUMAP(spat.merge.all, dims = 1:20)
# Saving on object in RData format
# save(spat.merge.all, file = "0.5_sct_spat.merge.all.RData")

#######################################################

DimPlot(spat.merge.all, reduction = "umap", group.by = c("ident"),label = TRUE)
DimPlot(spat.merge.all, reduction = "umap", group.by = c("orig.ident"))
DimPlot(spat.merge.all, reduction = "umap", split.by = c("orig.ident"))


```

```{r 011, dimreduction,  message=FALSE, ,width = 10,fig.height = 10}
markers <- FindAllMarkers(spat.merge.all, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
markers %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(spat.merge.all, features = top10$gene) + NoLegend()
```


```{r 012, dimreduction,  message=FALSE, ,width = 8,fig.height = 10}
SpatialDimPlot(spat.merge.all)

#Decision for subsetting 
cluster123A2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(0, 1, 2)), facet.highlight = TRUE, ncol = 3, images ="slice1")
cluster123B2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(0, 1, 2)), facet.highlight = TRUE, ncol = 3, images ="slice1_B2")
cluster123C2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(0, 1, 2)), facet.highlight = TRUE, ncol = 3, images ="slice1_C2")
cluster123D2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(0, 1, 2)), facet.highlight = TRUE, ncol = 3, images ="slice1_D2")

wrap_plots(cluster123A2, cluster123B2, cluster123C2, cluster123D2,nrow=4)

cluster345A2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(3, 4, 5)), facet.highlight = TRUE, ncol = 3, images ="slice1")
cluster345B2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(3, 4, 5)), facet.highlight = TRUE, ncol = 3, images ="slice1_B2")
cluster345C2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(3, 4, 5)), facet.highlight = TRUE, ncol = 3, images ="slice1_C2")
cluster345D2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(3, 4, 5)), facet.highlight = TRUE, ncol = 3, images ="slice1_D2")

wrap_plots(cluster345A2, cluster345B2, cluster345C2, cluster345D2,nrow=4)


cluster678A2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(6, 7, 8)), facet.highlight = TRUE, ncol = 3, images ="slice1")
cluster678B2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(6, 7, 8)), facet.highlight = TRUE, ncol = 3, images ="slice1_B2")
cluster678C2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(6, 7, 8)), facet.highlight = TRUE, ncol = 3, images ="slice1_C2")
cluster678D2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(6, 7, 8)), facet.highlight = TRUE, ncol = 3, images ="slice1_D2")

wrap_plots(cluster678A2, cluster678B2, cluster678C2, cluster678D2, nrow=4)

cluster91011A2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(9, 10, 11)), facet.highlight = TRUE, ncol = 3, images ="slice1")
cluster91011B2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(9, 10, 11)), facet.highlight = TRUE, ncol = 3, images ="slice1_B2")
cluster91011C2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(9, 10, 11)), facet.highlight = TRUE, ncol = 3, images ="slice1_C2")
cluster91011D2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(9, 10, 11)), facet.highlight = TRUE, ncol = 3, images ="slice1_D2")

wrap_plots(cluster91011A2, cluster91011B2, cluster91011C2, cluster91011D2, nrow=4)

# cluster121314A2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12, 13, 14)), facet.highlight = TRUE, ncol = 3, images ="slice1")
# cluster121314B2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12, 13, 14)), facet.highlight = TRUE, ncol = 3, images ="slice1_B2")
# cluster121314C2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12, 13, 14)), facet.highlight = TRUE, ncol = 3, images ="slice1_C2")
# cluster121314D2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12, 13, 14)), facet.highlight = TRUE, ncol = 3, images ="slice1_D2")

#wrap_plots(cluster121314A2, cluster121314B2, cluster121314C2, cluster121314D2, nrow=4)

# cluster12A2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12)), facet.highlight = TRUE, ncol = 1, images ="slice1")
# cluster12B2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12)), facet.highlight = TRUE, ncol = 1, images ="slice1_B2")
# cluster12C2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12)), facet.highlight = TRUE, ncol = 1, images ="slice1_C2")
# cluster12D2<-SpatialDimPlot(spat.merge.all, cells.highlight = CellsByIdentities(object = spat.merge.all, idents = c(12)), facet.highlight = TRUE, ncol = 1, images ="slice1_D2")
# 
# wrap_plots(cluster91011A2, cluster91011B2, cluster91011C2, cluster91011D2, nrow=4)

```


```{r 013, dimreduction,  message=FALSE, }

SpatialDimPlot(spat.merge.all, images ="slice1")
SpatialDimPlot(spat.merge.all, images ="slice1",alpha = 0)
SpatialDimPlot(spat.merge.all, images ="slice1_B2")
SpatialDimPlot(spat.merge.all, images ="slice1_B2",alpha = 0)
SpatialDimPlot(spat.merge.all, images ="slice1_C2")
SpatialDimPlot(spat.merge.all, images ="slice1_C2",alpha = 0)
SpatialDimPlot(spat.merge.all,  images ="slice1_D2")
SpatialDimPlot(spat.merge.all,  images ="slice1_D2",alpha = 0)

```

filtering will be applied when exclusion of folding regions

# For DGE we will use the log Normalised "Spatial" assay.

```{r 014, log_norm and save,  message=FALSE ,width = 10,fig.height = 8}
#Test<-spat.merge.all
DefaultAssay(spat.merge.all) <- "Spatial"
spat_merge_log_norm <- NormalizeData(spat.merge.all,assay = "Spatial", verbose = FALSE)
# ScaleData(spat_merge_log_norm)
#save(spat_merge_log_norm, file = "spat_merge_all_lognorm.RData")
```

